br/pkg/lightning/restore/meta_manager.go

```diff
@@ -586,23 +595,47 @@ func (m *dbTaskMetaMgr) CheckTaskExist(ctx context.Context) (bool, error) {
 	return exist, errors.Trace(err)
 }
 
-func (m *dbTaskMetaMgr) CheckClusterSource(ctx context.Context) (int64, error) {
+func (m *dbTaskMetaMgr) CheckTasksExclusively(ctx context.Context, action func(tx *sql.Tx, tasks []taskMeta) error) error {
 	conn, err := m.session.Conn(ctx)
 	if err != nil {
-		return 0, errors.Trace(err)
+		return errors.Trace(err)
 	}
 	defer conn.Close()
 	exec := &common.SQLWithRetry{
 		DB:     m.session,
 		Logger: log.L(),
 	}
-
-	source := int64(0)
-	query := fmt.Sprintf("SELECT SUM(source_bytes) from %s", m.tableName)
-	if err := exec.QueryRow(ctx, "query total source size", query, &source); err != nil {
-		return 0, errors.Annotate(err, "fetch task meta failed")
+	err = exec.Exec(ctx, "enable pessimistic transaction", "SET SESSION tidb_txn_mode = 'pessimistic';")
+	if err != nil {
+		return errors.Annotate(err, "enable pessimistic transaction failed")
 	}
-	return source, nil
+	return exec.Transact(ctx, "check tasks exclusively", func(ctx context.Context, tx *sql.Tx) error {
+		query := fmt.Sprintf("SELECT task_id, pd_cfgs, status, state, source_bytes, cluster_avail from %s FOR UPDATE", m.tableName)
```

Comment:
It seems that you have not add `cluster_avail` to task table?

Reply:
I previously add `cluster_avail` in action function, now update tasks in this function.